components/Chat/Chat.vue

```diff
@@ -197,22 +196,56 @@
         }
       },
 
+      async executeAction ( a ) {
+        if( a.insertMessage ) this.insertMessage( a.insertMessage );
+        if( a.saveToDb ) this.saveToDb( ...a.saveToDb );
+        if( a.forceFilter ) this.messages.filter( a.forceFilter );
+        if( a.changeStatOnGraph ) this.changeStatOnGraph( ...a.changeStatOnGraph );
+        if( a.chatServer ) {
+          this.chatServer = chatServers.get( a.chatServer );
+          await this.disconnectChat();
+          await this.connectToChat();
+        }
+      },
+
       async connectToChat () {
 
+        // TODO: collapse if
         await this.initChat();
```

Comment:
Idea was: `initChat()` returns the token/boolean, so you could inline the `if`. Petty, I know.

Reply:
Of that makes sense why it returns token now. smart.